final project.py

```diff
@@ -0,0 +1,161 @@
+                people_dict[id_search].printDetails()
+            else:
+                print("Error: ID " + id_search +" is not saved")
+        except:
+            errorNumber("ID", id_search)
+            return
+
+    def printAgesAverage(count, sum):
+        if count == 0:
+            print(count)
+        else:
+            result = sum / count
+            print(result)
+
+    def printAllNames(people_dict):
+        for index, id in enumerate(people_dict):
+            print(str(index) + ". " + people_dict[id].getName())
+
+    def printAllIDs(people_dict):
+        for index, id in enumerate(people_dict):
+            print(str(index) + ". " + id)
+
+    def printAllEntries(people_dict):
+        for index, id in enumerate(people_dict):
+            print(str(index) + ". ")
+            people_dict[id].printType()
+            people_dict[id].printMySelf()
+            
+    def printEntryByIndex(people_dict):
+        search_index = input("Please enter the index of the entry you want to print: ")
+        try:
+            if int(search_index) < len(people_dict):
+                for index, id in enumerate(people_dict):
+                    if search_index == str(index):
+                        people_dict[id].printType()
+                        people_dict[id].printDetails()
+                        break    
+            elif len(people_dict) > 0:
+                print("Error: Index out of range. The maximum index allowed is " + 
+                        str(len(people_dict) - 1))
+            else:
+                print("Error: The dictionary is empty")
+        except:
+            errorNumber("index", search_index)
+
+    def saveAllData(people_dict):
+        file_name = input("What is your output file name? ")
+        people_list = []
+        for id in people_dict.keys():
+            people_list.append(people_dict[id].getInfoDict())
+        df = pd.DataFrame(people_list)
+        close_file_flag = True
+        while close_file_flag:
+            try:
+                df.to_csv("C:\\Users\\Tehila\\Desktop\\Easy High-Tech\\PYTHON\\3\\" + file_name + ".csv", index=False)
+                print("The file " + file_name + " was saved successfully")
+                close_file_flag = False
+            except PermissionError:
+                print("Error: The file '" + file_name + ".csv' is open")
+                input_user = input("Please close the file and then press enter to continue")
+
+    users_dict = {}
+    user_choice = ""
+    sum_ages = 0
+    while True:
+        flag_y = False
+        flag_n = False
+        for option in Menu:
+            print("To " + option.name + " please press " + str(option.value))
+        user_choice = input("Please enter your choice: ")
+        try:
+            user_choice = Menu(int(user_choice))
+        except:
+            print("Option [" + user_choice + "] does not exist. Please try again")  
+            user_choice = input("Press Enter to continue ")

```

Comment:
טיפול יפה. אפשר לוותר על input("Press Enter to continue "), ובכל אופן לא צריך לשמור את זה בuser_choice